src/main/java/co/inventorsoft/academy/collections/model/Range.java

```diff
@@ -1,60 +1,177 @@
 package co.inventorsoft.academy.collections.model;
 
-import java.util.Collection;
-import java.util.Iterator;
-import java.util.Set;
+import java.util.*;
+import java.util.function.Function;
 
-public class Range<T> implements Set<T> {
+public class Range<T> implements Set<T> , Comparable {
```

Comment:
Why do you need Comparable for Set? Do you have to compare **Range**'s among ourselves?
Comparable is used for things, that are comparable between each other like two numbers.

Reply:
I thought I need Comparable for override method compareTo().

Reply:
Yes, but do you need compareTo at all for **Range**? You don't have to compare two range between them.
You can't say like Range is "higher" then another one. You can say that number 2 is higher than 1.